public/utils/find_last_index.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export const findLastIndex = <T>(
+  array: T[],
+  predicate: (value: T, index: number, array: T[]) => unknown
+) => {
+  if (array.length === 0) {
+    return -1;
+  }
+  const index = [...array].reverse().findIndex(predicate);
+  if (index === -1) {
+    return -1;
+  }
```

Comment:
why don't just search from last to first?

Reply:
Is that means write a for loop to search from end to start?

Reply:
Yes, that sounds like more straightforward and no additional array copy and reverse

Reply:
Make sense, will update later.